src/components/Modal.astro

```diff
@@ -0,0 +1,94 @@
+---
+interface Props {
+  modalContent: {
+    title?: string;
+    description?: string;
+  };
+}
+
+const {
+  modalContent: { title, description },
+} = Astro.props;
+---
+
+<div
+  id="modal"
+  class="fixed hidden top-1/2 left-1/2 max-w-2xl w-full -translate-x-1/2 -translate-y-1/2"
+  data-show="false"
+>
+  <div
+    class="mx-8 p-6 relative flex flex-col bg-sky-200 border-sky-400 border-2 md:p-10"
+  >
+    <div class="flex justify-center">
+      <button class="p-2 absolute w-12 h-12 top-0 right-0" id="icon-close">
+        <img src=`${import.meta.env.BASE_URL}/cross.svg` alt="cross icon" />
+        <span class="sr-only">Cross icon to close modal</span>
+      </button>
+      <slot name="title">
+        <h2 class="text-2xl md:text-4xl">{title}</h2>
+      </slot>
+    </div>
+    <div>
+      <slot>
+        <p class="mt-8 text-base overflow-auto max-h-[40vh] md:text-xl">
+          {description}
+        </p>
+      </slot>
+      <div class="mt-10 flex justify-between w-full">
+        <button class="btn" id="btn-close">close</button>
+        <button class="btn">confirm</button>
+      </div>
+    </div>
+  </div>
+</div>
+
+<script>
+  function modal() {
+    const modalContent: HTMLElement = document.querySelector("#modal")!;
+    const checkOfferBtn: any = document.querySelector("#btn-offer");
+    const closeModalBtn = modalContent.querySelector("#btn-close")!;
+    const closeModalIcon = modalContent.querySelector("#icon-close")!;
+    const main = document.querySelector("#main");
+
+    function close() {
+      modalContent.dataset.show = "false";
+      checkOfferBtn.disabled = false;
+      main?.classList.remove("blur");
+      document.removeEventListener("click", closeOutside, true);
+    }
+
+    function closeOutside(e) {
+      if (!modalContent.contains(e.target)) {
+        close();
+      }
+    }
+
+    function open() {
+      modalContent.dataset.show = "true";
+      checkOfferBtn.disabled = true;
+      main?.classList.add("blur");
+      document.addEventListener("click", closeOutside, true);
+    }
+
+    checkOfferBtn.addEventListener("click", open);
+
+    [closeModalBtn, closeModalIcon].forEach((closeModal) =>
+      closeModal.addEventListener("click", close)
+    );
```

Comment:
When https://github.com/bright-bounty/astro-lp/pull/14/files#r1605886993 is applied you can just iterate on the list
```suggestion
    closeModalBtns.forEach((closeModal) =>
      closeModal.addEventListener("click", close)
    );
```

Reply:
Changed